src/material/datepicker/date-range-input-parts.ts

```diff
@@ -35,14 +35,16 @@ import {
   MatDateFormats,
   ErrorStateMatcher,
   DateRange,
-  MatDateSelectionModel,
 } from '@angular/material/core';
 import {BooleanInput} from '@angular/cdk/coercion';
-import {MatDatepickerInputBase} from './datepicker-input-base';
+import {MatDatepickerInputBase, DateFilterFn} from './datepicker-input-base';
 
 /** Parent component that should be wrapped around `MatStartDate` and `MatEndDate`. */
-export interface MatDateRangeInputParent {
+export interface MatDateRangeInputParent<D> {
```

Comment:
why this interface instead of just using `MatDateRangeInput` directly?

Reply:
The base class reads some fields off the input and I didn't want to couple it to a particular type of input.

Reply:
Should we remove the underscore on these properties then? Since it's a public interface that people might want to implement? We can mark the properties `@docs-private` on `MatDateRangeInput` so they don't show up in the docs

Reply:
Then we'd have to either duplicate the properties or have getters which also isn't great. I named them this way so they align with what we already have on the datepicker input.

Reply:
Ah ok, lets just keep it then